sync-core/src/main/java/com/cloudant/sync/replication/BasicPushStrategy.java

```
@@ -315,7 +318,12 @@ private ItemsToPush missingRevisionsToJsonDocs(
 
     public String getReplicationId() {
         HashMap<String, String> dict = new HashMap<String, String>();
-        dict.put("source", this.sourceDb.getIdentifier());
+        try {
+            dict.put("source", this.sourceDb.getIdentifier());
+        } catch (DatastoreException e){
+            logger.log(Level.WARNING,"Failed to get ID using default",e);
+            dict.put("source","touchdb_defaultUUID");
+        }
```

Again, silently swallowing this seems like a Very Bad Idea.